quickstarts/deploy-vm/main.go

```diff
@@ -30,55 +24,55 @@ const (
 	parametersFile = "vm-quickstart-params.json"
 )
 
-var (
-	config = authInfo{ // Your application credentials
-		TenantID:               "", // Azure account tenantID
-		SubscriptionID:         "", // Azure subscription subscriptionID
-		ServicePrincipalID:     "", // Service principal appId
-		ServicePrincipalSecret: "", // Service principal password/secret
-	}
-
-	ctx = context.Background()
+// Information loaded from the authorization file to identify the client
+type clientInfo struct {
+	SubscriptionID string
+	VMPassword     string
+}
 
-	token *adal.ServicePrincipalToken
+var (
+	ctx        = context.Background()
+	clientData clientInfo
+	authorizer autorest.Authorizer
 )
 
-// Authenticate with the Azure services over OAuth, using a service principal.
+// Authenticate with the Azure services using file-based authentication
 func init() {
-	oauthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, config.TenantID)
-	if err != nil {
-		log.Fatalf("Failed to get OAuth config: %v\n", err)
-	}
-	token, err = adal.NewServicePrincipalToken(
-		*oauthConfig,
-		config.ServicePrincipalID,
-		config.ServicePrincipalSecret,
-		azure.PublicCloud.ResourceManagerEndpoint)
+	var err error
+	authorizer, err = auth.NewAuthorizerFromFile(azure.PublicCloud.ResourceManagerEndpoint)
```

Comment:
I would suggest to showcase `NewAuthorizerFromEnvironment()` as that is the recommended way to go.

Reply:
I should also mention that there is an entire how-to coming on authentication which __will__ recommend environment-based auth for most scenarios. When that's published the quickstart will link out to it explicitly as a document that features best practices.